app/workers/proactive_project_sync_worker.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ProactiveProjectSyncWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: :small, lock: :until_executed
+
+  PLATFORMS = %w[NPM Maven].freeze
+  DEFAULT_LIMIT = 1000
+
+  def perform(limit = DEFAULT_LIMIT)
+    projects_query
+      .limit(limit)
+      .each(&:async_sync)
+  end
+
+  def projects_query
+    Project
+      .distinct
+      .visible
+      .platform(PLATFORMS)
+      .where.associated(:repository_maintenance_stats)
+      .where("projects.last_synced_at IS NULL OR projects.last_synced_at < ?", last_synced_at_cutoff)
+      .order("projects.last_synced_at ASC NULLS FIRST")
+  end
+
+  def last_synced_at_cutoff
```

Comment:
WDYT about making this a constant instead of a public instance method?

Reply:
If it's a constant, wouldn't it get frozen to whatever the value is at load time? That's what I was avoiding